compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1245,12 +1245,16 @@ object Parsers {
               }
             })
 
-      if (in.token == STRINGPART)
-        nextSegment(in.offset + (if (isTripleQuoted) 3 else 1))
+      var offsetCorrection = if isTripleQuoted then 3 else 1
+      def offset = {
+        val result = in.offset + offsetCorrection
+        offsetCorrection = 0
+        result
+      }
       while (in.token == STRINGPART)
-        nextSegment(in.offset)
+        nextSegment(offset)
       if (in.token == STRINGLIT)
-        segmentBuf += literal(inPattern = inPattern, negOffset = in.offset, inStringInterpolation = true)
+        segmentBuf += literal(inPattern = inPattern, negOffset = offset, inStringInterpolation = true)
```

Comment:
Factoring out the `offset` is making the code harder to understand. If we inline `offset` and eliminate some redundant operation we can write it as:

```scala
      var offsetCorrection = if isTripleQuoted then 3 else 1
      while (in.token == STRINGPART)
        nextSegment(in.offset + offsetCorrection)
        offsetCorrection = 0
      if (in.token == STRINGLIT)
        segmentBuf += literal(inPattern = inPattern, negOffset = in.offset + offsetCorrection, inStringInterpolation = true)
```